components/style/properties/gecko.mako.rs

```diff
@@ -240,71 +240,40 @@ def set_gecko_property(ffi_name, expr):
     }
 </%def>
 
-<%def name="clear_color_flags(color_flags_ffi_name)">
-    % if color_flags_ffi_name:
-    self.gecko.${color_flags_ffi_name} &= !(structs::BORDER_COLOR_SPECIAL as u8);
-    % endif
-</%def>
-
-<%def name="set_current_color_flag(color_flags_ffi_name)">
-    % if color_flags_ffi_name:
-    self.gecko.${color_flags_ffi_name} |= structs::BORDER_COLOR_FOREGROUND as u8;
-    % else:
-    // FIXME(heycam): This is a Gecko property that doesn't store currentColor
-    // as a computed value.  These are currently handled by converting
-    // currentColor to the current value of the color property at computed
-    // value time, but we don't have access to the Color struct here.
-    // In the longer term, Gecko should store currentColor as a computed
-    // value, so that we don't need to do this:
-    // https://bugzilla.mozilla.org/show_bug.cgi?id=760345
-    warn!("stylo: mishandling currentColor");
-    % endif
-</%def>
-
-<%def name="get_current_color_flag_from(field)">
-    (${field} & (structs::BORDER_COLOR_FOREGROUND as u8)) != 0
-</%def>
-
-<%def name="impl_color_setter(ident, gecko_ffi_name, color_flags_ffi_name=None)">
+<%def name="impl_color_setter(ident, gecko_ffi_name, complex_color=True)">
     #[allow(unreachable_code)]
     #[allow(non_snake_case)]
     pub fn set_${ident}(&mut self, v: longhands::${ident}::computed_value::T) {
-        use cssparser::Color;
-        ${clear_color_flags(color_flags_ffi_name)}
-        let result = match v {
-            Color::CurrentColor => {
-                ${set_current_color_flag(color_flags_ffi_name)}
-                0
-            },
-            Color::RGBA(rgba) => convert_rgba_to_nscolor(&rgba),
-        };
+        % if complex_color:
+            let result = v.into();
+        % else:
+            use cssparser::Color;
+            let result = match v {
+                Color::RGBA(rgba) => convert_rgba_to_nscolor(&rgba),
+                Color::CurrentColor => 0,
```

Comment:
They can all be currentColor, I think that the code hasn't been updated yet to use complex colors for all of them (@heycam?)